src/main/java/org/folio/rest/impl/ProtectionHelper.java

```diff
@@ -0,0 +1,92 @@
+package org.folio.rest.impl;
+
+import static org.folio.orders.utils.ErrorCodes.ORDER_UNITS_NOT_FOUND;
+import static org.folio.orders.utils.ErrorCodes.USER_HAS_NO_PERMISSIONS;
+import static org.folio.orders.utils.HelperUtils.convertIdsToCqlQuery;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.HttpStatus;
+import org.folio.orders.rest.exceptions.HttpException;
+import org.folio.orders.utils.ProtectedOperationType;
+import org.folio.rest.jaxrs.model.AcquisitionsUnit;
+import org.folio.rest.jaxrs.model.AcquisitionsUnitCollection;
+import org.folio.rest.jaxrs.model.CompositePurchaseOrder;
+
+import io.vertx.core.Context;
+
+public class ProtectionHelper extends AbstractHelper {
+
+  private AcquisitionsUnitsHelper acquisitionsUnitsHelper;
+
+
+  public ProtectionHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(okapiHeaders, ctx, lang);
+    acquisitionsUnitsHelper = new AcquisitionsUnitsHelper(okapiHeaders, ctx, lang);
+  }
+
+  /**
+   * This method determines status of operation restriction based on unit IDs from {@link CompositePurchaseOrder}.
+   * @param unitIds list of unit IDs.
+   *
+   * @throws HttpException if user hasn't permissions or units not found
+   */
+  public CompletableFuture<Void> isOperationRestricted(List<String> unitIds, ProtectedOperationType operation) {
+    if (CollectionUtils.isNotEmpty(unitIds)) {
+      return getUnitsByIds(unitIds)
+        .thenCompose(units -> {
+          if (unitIds.size() == units.size()) {
```

Comment:
This check potentially might introduce problems. Let's imagine that order has 2 units assigned `A` and `B`. For some reason unit `B` has been deleted but order still references its id. User attempts to remove the unit from the order. This check won't pass and an error will be returned back to client i.e. there is no way to unassign such unit...

Reply:
I don't see an easy way to solve this here, maybe it's worth to put some logic to Acquisitions units API that will check if there are any records referring to this unit before deleting it...

Reply:
Looks like this is the only way to resolve potential issue....